### PR TITLE
feat: allow delete event-type platform atoms

### PR DESCRIPTION
**Pull request by ThyMinimalDev**
## What does this PR do?

Feat: allow deleting event-type in event type atom

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A-I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: packages/features/eventtypes/components/EventType.tsx [hunk 1]
```diff
@@ -62,10 +62,11 @@ export const EventType = (
     isPlatform?: boolean;
     tabName: (typeof tabs)[number];
     tabsNavigation: VerticalTabItemProps[];
+    allowDelete?: boolean;
   }
 ) => {
   const { formMethods, isPlatform, tabName } = props;
-  const { eventType, team, currentUserMembership, tabMap, isUpdating } = props;
+  const { eventType, team, currentUserMembership, tabMap, isUpdating, allowDelete = true } = props;
```
Comment (supalarry) [inline packages/features/eventtypes/components/EventType.tsx:69]: I think it's more clear if default true value is set in props type
```
allowDelete?: boolean = true;
```

File: packages/platform/atoms/hooks/event-types/private/useDeleteEventTypeById.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import http from "../../../lib/http";
+
+export const QUERY_KEY = "use-delete-event-by-id";
+export type UseDeleteEventTypeProps = {
+  onSuccess?: () => void;
+  onError?: (err: Error) => void;
+  onSettled?: () => void;
+};
+export const useDeleteEventTypeById = ({
+  onSuccess = () => {
```
Comment (supalarry) [inline packages/platform/atoms/hooks/event-types/private/useDeleteEventTypeById.ts:16]: Why empty onSuccess, onError and onSettled handlers?

**Review by supalarry — DISMISSED** [review]
Looks good with small notes

File: packages/features/eventtypes/components/EventType.tsx (hunk 1 above)
Comment (ThyMinimalDev) [reply packages/features/eventtypes/components/EventType.tsx:69]: agreed

**Review by Ryukemeister — APPROVED** [review]
Works well, really dope one!

**Pull request merged**